sentry-actix/src/lib.rs

```diff
@@ -645,6 +648,66 @@ mod tests {
         assert!(events.is_empty());
     }
 
+    /// Ensures client errors (4xx) returned by middleware are not captured.
+    #[actix_web::test]
+    async fn test_middleware_client_errors_discarded() {
+        let events = sentry::test::with_captured_events(|| {
+            block_on(async {
+                async fn hello_world() -> HttpResponse {
+                    HttpResponse::Ok().body("Hello, world!")
+                }
+
+                let app = init_service(
+                    App::new()
+                        .wrap_fn(|_, _| async {
+                            Err(actix_web::error::ErrorNotFound("Not found"))
+                                as Result<ServiceResponse<BoxBody>, _>
+                        })
+                        .wrap(Sentry::builder().with_hub(Hub::current()).finish())
+                        .service(web::resource("/test").to(hello_world)),
+                )
+                .await;
+
+                let req = TestRequest::get().uri("/test").to_request();
+                let res = app.call(req).await;
+                assert!(res.is_err());
+                assert!(res.unwrap_err().error_response().status().is_client_error());
```

Comment:
[nit] I would check the exact status code, not just that it is a client error here